cpp/src/arrow/compute/kernels/aggregate_basic.cc

```diff
@@ -894,6 +954,28 @@ const FunctionDoc mean_doc{
     {"array"},
     "ScalarAggregateOptions"};
 
+const FunctionDoc first_last_doc{"Compute the first and last values of an array",
+                                 ("Do not use this directly. This is internal and might"
+                                  "be removed in the future."),
```

Comment:
We can probably remove this warning.  I do believe that "always calculating the first and last in one pass" is not ideal when the user just wants the first or just wants the last (e.g. compared to min/max where you have to iterate the entire array no matter what).

However, I do think there will still be scenarios where a user will want both the first and last in a single pass.  So let's go ahead and assume this function is here to stay.

Reply:
Sounds good. Will update.

Reply:
Updated